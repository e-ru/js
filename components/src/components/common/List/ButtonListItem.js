import React from "react";
import PropTypes from "prop-types";

import { makeStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";

const useStyles = makeStyles(theme => ({
  nested: {
    paddingLeft: theme.spacing(4),
  },
}));

const ButtonListItem = ({ subItem, title, icon, onClickHanlder }) => {
  const classes = useStyles();
  return (
    <ListItem button className={subItem ? classes.nested : ""} key={title} onClick={onClickHanlder && onClickHanlder}>
      <ListItemIcon>{icon}</ListItemIcon>
      <ListItemText primary={title} />
    </ListItem>
  );
};

ButtonListItem.defaultProps = {
  onClickHanlder: null,
  subItem: false,
};

ButtonListItem.propTypes = {
  title: PropTypes.string.isRequired,
  icon: PropTypes.object.isRequired,
  subItem: PropTypes.bool,
  onClickHanlder: PropTypes.func,
};

export default ButtonListItem;
