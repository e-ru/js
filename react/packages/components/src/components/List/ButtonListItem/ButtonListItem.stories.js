import React from "react";
import { storiesOf } from "@storybook/react";

import HomeIcon from "@material-ui/icons/Home";
import ButtonListItem from "./ButtonListItem";

storiesOf("ButtonListItem", module)
  .add("main item", () => (
    <ButtonListItem
      subItem={false}
      title="main"
      icon={<HomeIcon />}
      onClickHandler={() => console.log("onClickHandler clicked")}
    />
  ))
  .add("sub item", () => (
    <ButtonListItem
      subItem
      title="main"
      icon={<HomeIcon />}
      onClickHandler={() => console.log("onClickHandler clicked")}
    />
  ));
