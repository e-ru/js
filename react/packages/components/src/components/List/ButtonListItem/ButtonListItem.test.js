import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import ButtonListItem from "./ButtonListItem";

configure({ adapter: new Adapter() });

function setup() {
  const props = {
    title: "test_title",
    icon: {},
  };

  const wrapper = shallow(<ButtonListItem {...props} />);

  return {
    wrapper,
  };
}

describe("<ButtonListItem />", () => {
  test("renders", () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
