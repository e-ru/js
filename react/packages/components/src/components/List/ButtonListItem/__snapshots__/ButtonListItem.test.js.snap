// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonListItem /> renders 1`] = `ShallowWrapper {}`;
